modules/core/src/views/view-manager.js

```diff
@@ -153,12 +165,18 @@ export default class ViewManager {
 
   // Update the view descriptor list and set change flag if needed
   // Does not actually rebuild the `Viewport`s until `getViewports` is called
-  setViews(views) {
+  _setViews(views) {
     // DEPRECATED: Ensure any "naked" Viewports are wrapped in View instances
     views = flatten(views, {filter: Boolean}).map(
       view => (view instanceof Viewport ? new View({viewportInstance: view}) : view)
     );
 
+    views.forEach(view => {
+      if (view.controller && !this.controllers[view.id]) {
```

Comment:
Is there a need to get rid of unused controllers. Maybe a comment describing why we do what we do, either way?